### PR TITLE
updates buf and regenerates

### PR DESCRIPTION
**Pull request by vroldanbet**
The buf version used to generate proto/grpc code is relatively old, this bumps the version and regenerates

### CODE REVIEW

File: .github/workflows/automatic-api-update.yaml [hunk 1]
```diff
@@ -32,11 +32,11 @@ jobs:
       - name: "Install ruby"
         uses: "ruby/setup-ruby@v1"
         with:
-          ruby-version: "2.7.3"
+          ruby-version: "3.2.2"
```
Comment (jzelinskie) [inline .github/workflows/automatic-api-update.yaml:35]: I think we were previously targeting an older but still commonly used version of Ruby and not the latest. @samkim do you remember how we got this version?

Comment (vroldanbet) [reply .github/workflows/automatic-api-update.yaml:35]: I can rollback this change, I though it was worth starting to test 3.X

Comment (vroldanbet) [reply .github/workflows/automatic-api-update.yaml]: @jzelinskie reverted: https://github.com/authzed/authzed-rb/pull/76/commits/1adf79dba55e472dcec8fb12ec046de663c39914

**Review by jzelinskie — APPROVED** [review]

**Comment by jzelinskie** [comment]
we can figure out ideal ruby versions as a follow up

**Pull request merged**